src/cmap/mod.rs

```diff
@@ -51,144 +58,35 @@ impl From<Arc<ConnectionPoolInner>> for ConnectionPool {
     }
 }
 
-/// A struct used to manage the creation, closing, and storage of connections for a
-/// `ConnectionPool`.
-#[derive(Debug)]
-struct ConnectionManager {
```

Comment:
I ended up removing just removing this type and consolidating everything back onto `ConnectionPoolInner`, with the `Vec` of available connections being the only thing guarded by the lock now. This is basically how it originally was before I added the `ConnectionManager`. The initial goal of `ConnectionManager` was to simplify the considerations around concurrency and state updates of the pool, but it ended up being too restrictive to be useful. I also think that now that the spec has been updated to be more clear about these state changes, the simplicity that `ConnectionManager` got us isn't as worthwhile. (e.g. use of `PendingConnection` helps describe these state changes at the type level).